### ISSUE TITLE
Search query: Found more than 1 matching build

### ISSUE THREAD

**Issue by Achsuthan** [fastlane/fastlane#22248]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I was trying to upload the iOS build using the script, getting the error always saying 

ERROR [2024-09-26 13:01:57.68]: fastlane finished with errors
Looking for related GitHub issues on fastlane/fastlane...
Search query: Found more than 1 matching build: 
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING

##### Command executed

api_key = app_store_connect_api_key(
      key_id: ENV["APP_STORE_CONNECT_API_KEY_ID"],
      issuer_id: ENV["APP_STORE_CONNECT_ISSUER_ID"],
      key_filepath: ENV["APP_STORE_CONNECT_API_KEY_PATH"],
      duration: 1200,
      in_house: false
    )

    puts "Debug: API Key object created successfully"
    puts "Debug: App Identifier: #{ENV['ITUNES_CONNECT_APP_ID']}"

    # Upload to TestFlight
    begin
      upload_to_testflight(
        api_key: api_key,
        ipa: ENV["IPA_PATH"],
        app_version: ENV["APP_VERSION"],
        skip_waiting_for_build_processing: false,
        apple_id: ENV["ITUNES_CONNECT_APP_ID"],
	changelog: ENV["TESTFLIGHT_CHANGELOG"],
	distribute_external: true,
	groups: ["TESTING"]
	
      )

##### Complete output when running fastlane, including the stack trace and command used

INFO [2024-09-26 13:00:23.40]: Successfully uploaded the new binary to App Store Connect
INFO [2024-09-26 13:00:23.40]: Creating authorization token for App Store Connect API
DEBUG [2024-09-26 13:00:23.42]: App Platform (ios)
INFO [2024-09-26 13:00:24.31]: Waiting for processing on... app_id: [REDACTED], app_version: 4.3.0, build_version: 1259, platform: IOS
WARN [2024-09-26 13:00:24.78]: Read more information on why this build isn't showing up yet - https://github.com/fastlane/fastlane/issues/14997
INFO [2024-09-26 13:00:24.78]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO [2024-09-26 13:00:55.18]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO [2024-09-26 13:01:25.67]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
ERROR [2024-09-26 13:01:57.67]: Error uploading to TestFlight: Found more than 1 matching build: 
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING
WARN [2024-09-26 13:01:57.67]: Lane Context:
INFO [2024-09-26 13:01:57.67]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta"}
ERROR [2024-09-26 13:01:57.67]: Found more than 1 matching build: 
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING
INFO [2024-09-26 13:01:57.67]: Successfully generated documentation at path '/Users/vagrant/git/fastlane/README.md'
+------------------------------------------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | default_platform          | 0           |
| 2    | app_store_connect_api_key | 0           |
| 💥   | upload_to_testflight      | 183         |
+------+---------------------------+-------------+
ERROR [2024-09-26 13:01:57.68]: fastlane finished with errors
Looking for related GitHub issues on fastlane/fastlane...
Search query: Found more than 1 matching build: 
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING
URL: https://api.github.com/search/issues?q=Found%20more%20than%201%20matching%20build%3A%20%0A4.3.0%281259%29%20for%20IOS%20-%20PROCESSING%0A4.3.0%281259%29%20for%20IOS%20-%20PROCESSING+repo:fastlane/fastlane
➡️  Improve Beta Setup
    https://github.com/fastlane/fastlane/pull/1 [closed] 1 💬
    04 Feb 2017
🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:124:in `matching_build': [!] Found more than 1 matching build:  (FastlaneCore::BuildWatcherError)
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:37:in `block in wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:36:in `loop'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:36:in `wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/pilot/lib/pilot/build_manager.rb:125:in `wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/pilot/lib/pilot/build_manager.rb:81:in `upload'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:34:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:19:in `block (2 levels) in parsing_binding'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/lane.rb:41:in `call'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/command_line_handler.rb:34:in `handle'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:363:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/bin/fastlane:23:in `<top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2/bin/fastlane:25:in `load'
	from /Users/vagrant/.asdf/installs/ruby/3.2/bin/fastlane:25:in `<main>'
/Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:124:in `matching_build': Found more than 1 matching build:  (FastlaneCore::BuildWatcherError)
4.3.0(1259) for IOS - PROCESSING
4.3.0(1259) for IOS - PROCESSING
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:37:in `block in wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:36:in `loop'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/build_watcher.rb:36:in `wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/pilot/lib/pilot/build_manager.rb:125:in `wait_for_build_processing_to_be_complete'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/pilot/lib/pilot/build_manager.rb:81:in `upload'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:34:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:19:in `block (2 levels) in parsing_binding'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/lane.rb:41:in `call'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/command_line_handler.rb:34:in `handle'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:363:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/fastlane-2.220.0/bin/fastlane:23:in `<top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2/bin/fastlane:25:in `load'
	from /Users/vagrant/.asdf/installs/ruby/3.2/bin/fastlane:25:in `<main>'

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by iblacksun** [comment]
same here

**Comment by trmquangbiz** [comment]
same here
7.4.0(16) for IOS - PROCESSING
7.4.0(16) for IOS - PROCESSING

**Comment by Cooper-Han** [comment]
Found more than 1 matching build: 
1.33.0(21382) for IOS - PROCESSING
1.33.0(21382) for IOS - PROCESSING

**Comment by rsZWang** [comment]
same here

**Comment by xiaozhi747** [comment]
+1

**Comment by DoctorCobweb** [comment]
something todo with app compliance failed for the uploaded beta version in appstore connect testflight, maybe?

each time i built, it would upload but require filling out a compliance survey for i) encryption protocols used and ii) if app was used in france

**Comment by Cooper-Han** [comment]
> something todo with app compliance failed for the uploaded beta version in appstore connect testflight, maybe?
> 
> each time i built, it would upload but require filling out a compliance survey for i) encryption protocols used and ii) if app was used in france

I think it must be an error with Apple’s services. In fact, the version of this build is already visible in App Store Connect.

**Comment by xiaozhi747** [comment]
I found that this problem only occurs when uploading ipa and distributing at the same time. At present, I can bypass it by performing the operations of uploading ipa and distributing separately.

My temporary solution is to call upload_to_testflight twice. The first call can successfully upload the ipa, but there will be an error of distributing failure, which needs to be ignored; the second call adds `distribute_only: true` to complete the distribution.

**Comment by bianka-l** [comment]
+1, all my builds are failing with the same error

**Comment by phasedarray** [comment]
did you guys find that all the reported build versions end with '0', like 7.4.0， 1.33.0 .
So the walk-around is to change version to 7.4.1， 1.33.1.
🎉

**Comment by OguzYuuksel** [comment]
We are having same issue..

**Comment by patelnishantk** [comment]
+1 Same issue

**Comment by skiryuk** [comment]
++++1

**Comment by nklausni** [comment]
Same here +1

**Comment by simone-smith** [comment]
We're having the same issue

**Comment by Robin-Gupta1** [comment]
I am also facing the same issue at my end.

**Comment by yakupemeksiz** [comment]
Same here +1

**Comment by ugurcanatas** [comment]
> I can also confirm that we are having a similar problem.
> 
> Builds are uploaded to TestFlight. However CircleCI job basically fails because of the described issue.
> 
> Also as an additional info: changelog & localized_build_info is not visible on test flight builds

This is going to sound weird but out app version was ending with 0, as explained in [this comment](https://github.com/fastlane/fastlane/issues/22248#issuecomment-2376576705)

we changed it to end with 1 and now I can see the `changelog` as well as `localized_build_info` on testflight. Builds are successful on CI as well

**Comment by tuan-to** [comment]
> I found that this problem only occurs when uploading ipa and distributing at the same time. At present, I can bypass it by performing the operations of uploading ipa and distributing separately.
> 
> My temporary solution is to call upload_to_testflight twice. The first call can successfully upload the ipa, but there will be an error of distributing failure, which needs to be ignored; the second call adds `distribute_only: true` to complete the distribution.
> 
> `distribute_only` will distributing the latest version. You can add `app_version` and `build_number` parameters to specify the version.

I'm doing the same. It was OK before but it become error from yesterday.
Here is error log with `distribute_only: false`:
Called from Fastfile at line 68
```
    66:	  desc 'Upload build to Apple App Store Connect.'
    67:	  lane :uploadBuildToAppStoreConnect do
 => 68:	    upload_to_testflight(
    69:	      ipa: ENV["PILOT_IPA"],
    70:	      distribute_only: false,
```
Found more than 1 matching build: 
1.0.0(7) for IOS - PROCESSING
1.0.0(7) for IOS - PROCESSING

**Comment by tuan-to** [comment]
> did you guys find that all the reported build versions end with '0', like 7.4.0， 1.33.0 . **So the walk-around is to change version to 7.4.1， 1.33.1.（Verified）** 🎉

really? But this is mys first build to App Store, so version should be 1.0.0, right? It is weird for first version to be 1.0.1.

**Comment by grEvenX** [comment]
@tuan-to This is a work-around for a bug that seemingly started happening today. Probably related to something Apple has changed on their side. If you are currently blocked by this issue, you can apply the workaround (or any other workaround, like manually performing whatever your build pipeline does) of your choice or choose to wait for a fix.

**Comment by justincbeck** [comment]
FWIW, me too. ¯\_(ツ)_/¯

**Comment by psjostrom** [comment]
Going strong with a +1 here

**Comment by jonatasteixeira** [comment]
Same here!

**Comment by fabiendem** [comment]
Is a maintainer able to pinpoint which URL of the App Store Connect API is used and determine if it's not returning accurate results?

If we have this we can report the problem to Apple maybe?
- https://developer.apple.com/documentation/appstoreconnectapi
- https://forums.developer.apple.com/forums/tags/app-store-connect-api

**Comment by fabiendem** [comment]
Actually, for those who reported the issue, which version of fastlane are you using?
Logs in the initial message mention `2.220.0` while the latest version `2.222.0` 
https://github.com/fastlane/fastlane/releases

**Comment by fabiendem** [comment]
Are you using Bitrise?

Here I see that our script which is installing fastlane 2.221.1 is failing.
So Bitrise fallbacks to 2.220.0.


```
+ bundle exec pod install --repo-update --project-directory=ios/
bundler: failed to load command: pod (/Users/vagrant/.asdf/installs/ruby/3.2.4/bin/pod)
/Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/definition.rb:540:in `materialize': Could not find cocoapods-1.15.2, activesupport-7.0.8.4, fastlane-2.221.1, fastlane-plugin-checks-0.2.1, addressable-2.8.7, claide-1.1.0, cocoapods-core-1.15.2, cocoapods-deintegrate-1.0.5, cocoapods-downloader-2.1, cocoapods-plugins-1.0.0, cocoapods-search-1.0.1, cocoapods-trunk-1.6.0, cocoapods-try-1.2.0, colored2-3.1.2, escape-0.0.4, fourflusher-2.3.1, gh_inspector-1.1.3, molinillo-0.8.0, nap-1.1.0, ruby-macho-2.5.1, xcodeproj-1.25.0, concurrent-ruby-1.3.4, i18n-1.14.5, minitest-5.25.0, tzinfo-2.0.6, CFPropertyList-3.0.7, artifactory-3.0.17, aws-sdk-s3-1.158.0, babosa-1.0.4, colored-1.2, commander-4.6.0, dotenv-2.8.1, emoji_regex-3.2.3, excon-0.111.0, faraday-1.10.3, faraday-cookie_jar-0.0.7, faraday_middleware-1.2.0, fastimage-2.3.1, google-apis-androidpublisher_v3-0.54.0, google-apis-playcustomapp_v1-0.13.0, google-cloud-env-1.6.0, google-cloud-storage-1.47.0, highline-2.0.3, http-cookie-1.0.7, json-2.7.2, jwt-2.8.2, mini_magick-4.13.2, multipart-post-2.4.1, naturally-2.2.1, optparse-0.5.0, plist-3.7.1, rubyzip-2.3.2, security-0.1.5, simctl-1.6.10, terminal-notifier-2.0.0, terminal-table-3.0.2, tty-screen-0.8.2, tty-spinner-0.9.3, word_wrap-1.0.0, xcpretty-0.3.0, xcpretty-travis-formatter-1.0.1, googleauth-1.8.1, public_suffix-4.0.7, algoliasearch-1.27.5, fuzzy_match-2.0.4, netrc-0.11.0, typhoeus-1.4.1, atomos-0.1.3, nanaimo-0.3.0, rexml-3.3.6, base64-0.2.0, nkf-0.2.0, aws-sdk-core-3.201.5, aws-sdk-kms-1.88.0, aws-sigv4-1.9.1, faraday-em_http-1.0.0, faraday-em_synchrony-1.0.0, faraday-excon-1.1.0, faraday-httpclient-1.0.1, faraday-multipart-1.0.4, faraday-net_http-1.0.2, faraday-net_http_persistent-1.2.0, faraday-patron-1.0.0, faraday-rack-1.0.0, faraday-retry-1.0.3, google-apis-core-0.11.3, digest-crc-0.6.5, google-apis-iamcredentials_v1-0.17.0, google-apis-storage_v1-0.31.0, google-cloud-core-1.7.1, mini_mime-1.1.5, domain_name-0.6.20240107, unicode-display_width-2.5.0, tty-cursor-0.7.1, rouge-2.0.7, multi_json-1.15.0, os-1.1.4, signet-0.19.0, httpclient-2.8.3, ethon-0.16.0, strscan-3.1.0, aws-eventstream-1.3.0, aws-partitions-1.964.0, jmespath-1.6.2, representable-3.2.0, retriable-3.1.2, rake-13.2.1, google-cloud-errors-1.4.0, ffi-1.17.0, declarative-0.0.20, trailblazer-option-0.1.2, uber-0.1.0 in locally installed gems (Bundler::GemNotFound)
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/definition.rb:200:in `specs'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/definition.rb:266:in `specs_for'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/runtime.rb:18:in `setup'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler.rb:162:in `setup'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/setup.rb:23:in `block in <top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/ui/shell.rb:159:in `with_level'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/ui/shell.rb:111:in `silence'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/setup.rb:23:in `<top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli/exec.rb:56:in `require_relative'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli/exec.rb:56:in `kernel_load'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli/exec.rb:23:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli.rb:492:in `exec'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli.rb:34:in `dispatch'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/cli.rb:28:in `start'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/bundler-2.4.19/libexec/bundle:37:in `block in <top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/3.2.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/lib/ruby/gems/3.2.0/gems/bundler-2.4.19/libexec/bundle:29:in `<top (required)>'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/bin/bundle:25:in `load'
	from /Users/vagrant/.asdf/installs/ruby/3.2.4/bin/bundle:25:in `<main>'
+ exit_code=1
```

**Comment by moux2003** [comment]
same here with 2.222

```
/usr/local/lib/ruby/gems/3.0.0/gems/fastlane-2.222.0/fastlane_core/lib/fastlane_core/build_watcher.rb:124:in 'matching_build': [!] Found more than 1 matching build:  (FastlaneCore::BuildWatcherError)
13.0.0(1433) for IOS - PROCESSING
13.0.0(1433) for IOS - PROCESSING
```

**Comment by fabiendem** [comment]
Okay, so it's not a fastlane version issue 🤔

**Comment by moux2003** [comment]
I'm wondering why it should be an issue to have multiple builds with same number (it's probably related to an API issue, App store connect is so unreliable these past months..), shouldn't we consider just matching all values together ?

**Comment by moux2003** [comment]
`wait_for_build_processing_to_be_complete` can take a `select_latest` and maybe it should be given `true` from `UploadToTestflightAction` ? not sure who's decision it is 😃

**Comment by fabiendem** [comment]
Yeah, we need a maintainer a bit more familiar with the codebase to help 🙏

**Comment by phasedarray** [comment]
> > I can also confirm that we are having a similar problem.
> > Builds are uploaded to TestFlight. However CircleCI job basically fails because of the described issue.
> > Also as an additional info: changelog & localized_build_info is not visible on test flight builds
> 
> This is going to sound weird but out app version was ending with 0, as explained in [this comment](https://github.com/fastlane/fastlane/issues/22248#issuecomment-2376576705)
> 
> we changed it to end with 1 and now I can see the `changelog` as well as `localized_build_info` on testflight. Builds are successful on CI as well

if you look into the related code in fastlane, you will find they did some tricky thing about app versions with and without 0, like 4.9 and 4.9.0 (maybe caused by an Apple server bug). So i guess Apple fixed it these days, which broke the fastlane tricky code.

**Comment by TheMetalCode** [comment]
> if you look into the related code in fastlane, you will find they did some tricky thing about app versions with and without 0, like 4.9 and 4.9.0 (maybe caused by an Apple server bug). So i guess Apple fixed it these days, which broke the fastlane tricky code.
> So as long as we change app version to any one that doesn’t end with 0, it won’t trigger the tricky code and everything works fine.

I'm a bit skeptical about this - our app versions _always_ end in `.0` unless it's a hotfix/patch. Most of our releases bump minor version. We have not seen this behavior until just now like almost everyone else on this thread.

**Comment by ugurcanatas** [comment]
@TheMetalCode that's most likely because this is a recent fix on apple's side. Maybe yesterday because we started seeing this today from the morning. 

> if you look into the related code in fastlane ...
@phasedarray could you link the related code if possible?

**Comment by Him-Sharma** [comment]
I started seeing this error now, I am wondering if fastlane was uploading multiple times
`ERROR: [ContentDelivery.Uploader] Asset validation failed (90382) Upload limit reached. `

**Comment by phasedarray** [comment]
> @TheMetalCode that's most likely because this is a recent fix on apple's side. Maybe yesterday because we started seeing this today from the morning.
> 
> > if you look into the related code in fastlane ...
> 
> @phasedarray could you link the related code if possible?

I’m not at work now and don’t have dev environment.But you can follow the error message in which there’s a path to upload_to_testflight.rb. And see line 34 and comments above.then you’ll realize all things

**Comment by GriArS** [comment]
Same here! Thank God, I thought I broke something while reconfiguring the CI, lol

**Comment by TheMetalCode** [comment]
https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/build_watcher.rb#L118

Maybe I'm being too simplistic but I wonder if that line could be:

```ruby
matched_builds = version_matches.map(&:builds).flatten.uniq
```

I would definitely want to understand more about why we're getting two builds in that array that say the same version/buildnum info, but the fact of the matter is there's not actually two builds, it's listing the same one twice.

**Comment by lodibrahim** [comment]
We are also facing same issue

**Comment by roffelund** [comment]
> Here is the related PR #18547
> 
> > Apple sees versions with as X.Y and X.Y.0 as the same. This can lead to issues when waiting for build processing as the queryable version format is the type that was first uploaded.
> > Example: If first version was X.Y and next uploads were X.Y.0, the API returns nothing when querying for X.Y.0 but will for X.Y
> > This will lead to an infinite poll as nothing will ever appear for X.Y.0.
> 
> Which probably explains the situation if Apple fixed something on their side.
> 
> EDIT:
> 
> If anyone wants to compare the docs: (not sure I am looking at the correct place)
> 
> Here is the archived docs: https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364
> 
> Here is the current apple docs https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring/
> 
> Maybe if there's a change, it is visible in the docs.

Played around a bit on a fork. Havent been able to test yet. But if someone has time to continue :) 
https://github.com/fastlane/fastlane/pull/22254

**Comment by mikeger** [comment]
This change is fixing the issue for me: https://github.com/mikeger/fastlane/commit/e83f7698a0430135f871e45be8f800ed819d7ab1

**Comment by TheMetalCode** [comment]
Can confirm https://github.com/fastlane/fastlane/pull/22254 appears to address the issue!

I noticed an oddity in the logs, will comment further on https://github.com/fastlane/fastlane/pull/22254

**Comment by JesusRamp** [comment]
Setting  `skip_waiting_for_build_processing: true`, bypass the issue, at least for me.
 
 When the option is set to false, it does a call to Apple every x min to verify if "the build to show up in the build list".

**Comment by TheMetalCode** [comment]
@JesusRamp unfortunately `skip_waiting_for_build_processing: true` is not an option if you're including release notes in the `pilot` call.

**Comment by eppisapiafsl** [comment]
`skip_waiting_for_build_processing: true`  disables distribution to external tester 😞

**Comment by pacocd** [comment]
Same here

**Comment by paulocoutinhox** [comment]
+1 and this too: https://github.com/fastlane/fastlane/issues/22247

**Comment by peter-flexai** [comment]
+1

**Comment by jjkaufman** [comment]
Same issue here

**Comment by joshdholtz** [comment]
✨ Official Response

Woah! Sorry folks, just saw this issue 😬 Taking a look at the conversation now and will try to get a fix out quickly

**Comment by bz-Mega** [comment]
Same issue here beginning from yesterday 😅

**Comment by shion-koga** [comment]
Same here biginning from yesterday.

**Comment by rumaniel** [comment]
+1 to subscribe this thread.

**Comment by joshdholtz** [comment]
✨ Official Response 2

Is anybody around to test out my fix? Its #22256 🙏

**Comment by grEvenX** [comment]
@joshdholtz Started a build with it now. Will hopefully know in ~20 minutes if it works.

**Comment by ugurcanatas** [comment]
> ✨ Official Response 2
> 
> Is anybody around to test out my fix? Its #22256 🙏
> 
> Update `Gemfile` and run `bundle install`, `bundle update fastlane`, or `bundle update`
> 
> ```ruby
> gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "fix-duplicate-builds-in-build-watcher"
> ```
> 
> If this works, **please let me know** and I'll get a build out immediately 💪

Started builds as well 🤞

**Comment by Baza207** [comment]
Running ours too, though it will take longer than 20min 🙈

**Comment by grEvenX** [comment]
@joshdholtz I can confirm it works 👌 Thanks for the quick response.

<img width="969" alt="image" src="https://github.com/user-attachments/assets/c69bd7e2-1c42-47a3-b7eb-f21031f7ee04">

**Comment by joshdholtz** [comment]
> @joshdholtz I can confirm it works 👌 Thanks for the quick response.
> 
> <img alt="image" width="969" src="https://private-user-images.githubusercontent.com/26375/371407525-c69bd7e2-1c42-47a3-b7eb-f21031f7ee04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0MTY1NjMsIm5iZiI6MTcyNzQxNjI2MywicGF0aCI6Ii8yNjM3NS8zNzE0MDc1MjUtYzY5YmQ3ZTItMWM0Mi00N2EzLWI3ZWItZjIxMDMxZjdlZTA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDA1NTEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyYTI0Y2ZkMGY5OWJmMjlkYTU0MzRjMzM1OTcxNzQyYzQ4N2QwZDhmY2M1MzNkOTY4OTRiNDc0NGExYmRjMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.asdqjBWvqNRKlF8ZcyUeN3-gHb1ct3KqiNh_cMCOj9E">

@grEvenX Awesome news! I will get that release started 💪

**Comment by joshdholtz** [comment]
✨ Official Response 3

Making a new release right now! _Should_ be up within 30-ish minutes but I'll reply with new version here when its up

**Comment by ugurcanatas** [comment]
I can also confirm that fix works with having patch version as 0 at the end @joshdholtz

**Comment by joshdholtz** [comment]
✨ Official Response 4

Version [2.223.1](https://github.com/fastlane/fastlane/releases/tag/2.223.1) is out and live on RubyGems 🚀 💪 

I'm off to bed now but I'll check back in the morning to see if anything else popped up but... hopefully not 😇

**Comment by cong-toquang** [comment]
> ✨ Official Response 4
> 
> Version [2.223.1](https://github.com/fastlane/fastlane/releases/tag/2.223.1) is out and live on RubyGems 🚀 💪
> 
> ℹ️ The Homebrew version should be out in SOME amount of hours but I do not know exactly
> 
> I'm off to bed now but I'll check back in the morning to see if anything else popped up but... hopefully not 😇

Thank you very much. Confirm it worked!

**Comment by charleyoliveira** [comment]
+1 same issue here

**Comment by Zeropointer11** [comment]
thanks the fix is working

**Comment by jedai7** [comment]
I updated fast lane  to 2.224.0
fastlane --version 
fastlane installation at path:
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.224.0/bin/fastlane

update my gemfile.lock from fastlane (2.213.0) to fastlane (2.224.0). But i see this log in my pipeline "Using fastlane 2.214.0" and i still have issue with Found more than 1 matching build:  (FastlaneCore::BuildWatcherError)
Help pls

**Comment by grEvenX** [comment]
@jedai7 Then you are not using the correct version on your CI which is a different issue. You need to check your CI pipeline script to understand how you are invoking fastlane, and possibly what version is being used in your PATH ENV variable.
If you are using bundler, the CI would normally use `bundle install` and `bundle exec fastlane` to run the fastlane version specified by your gemfile.lock.
